x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/VersionParityTests.java

```diff
@@ -48,27 +47,31 @@ public void testExceptionThrownOnIncompatibleVersions() throws IOException, SQLE
                     + ClientVersion.CURRENT.majorMinorToString()
                     + " or newer; attempting to connect to a server "
                     + "version "
-                    + versionString,
+                    + version,
                 ex.getMessage()
             );
-        } while (version.compareTo(firstVersion) > 0);
+        }
     }
 
     public void testNoExceptionThrownForCompatibleVersions() throws IOException {
         String url = JdbcConfiguration.URL_PREFIX + webServerAddress();
-        Version version = Version.CURRENT;
-        try {
-            do {
+        List<SqlVersion> afterVersionCompatibility = new ArrayList<>(SqlVersions.getAllVersions().size());
+        SqlVersions.getAllVersions()
+            .stream()
+            .filter(v -> v.onOrAfter(INTRODUCING_VERSION_COMPATIBILITY))
+            .forEach(afterVersionCompatibility::add);
```

Comment:
nit - this could be `List<SqlVersion> compat = ....collect(Collectors.toCollection(ArrayList::new))`

Reply:
Thanks, fixed.